src/App.js

```diff
@@ -92,9 +101,49 @@ function App() {
     removeUserInfoFromCookies();
   }
 
+  function handleSuccessfulSignIn(userInfo, sessionId) {
+    console.log("HANDLE SUCCESSFUL SIGN IN");
+    api.session = sessionId;
+    console.log("Session: " + api.session);
+    LocalStorage.setSession(api.session);
+    LocalStorage.setUserInfo(userInfo);
+
+    // TODO: Should this be moved to Settings.loadUsrePreferences?
+    api.getUserPreferences((preferences) => {
+      SessionStorage.setAudioExercisesEnabled(
+        preferences["audio_exercises"] === undefined ||
+          preferences["audio_exercises"] === "true",
+      );
+    });
+
+    // Cookies are the mechanism via which we share a login
+    // between the extension and the website
+    saveUserInfoIntoCookies(userInfo, api.session);
+    let newUserValue = {
+      session: api.session,
+      name: userInfo.name,
+      learned_language: userInfo.learned_language,
+      native_language: userInfo.native_language,
+      is_teacher: userInfo.is_teacher,
+      is_student: userInfo.is_student,
+    };
+
+    console.log("setting new user value: ");
+    console.dir(newUserValue);
+    setUser(newUserValue);
+
+    /* If a redirect link exists, uses it to redirect the user,
+        otherwise, uses the location from the function argument. */
+    handleRedirectLinkOrGoTo("/articles");
+  }
+
   //Setting up the routing context to be able to use the cancel-button in EditText correctly
   const [returnPath, setReturnPath] = useState("");
 
+  if (userData === undefined) {
+    return <div>Loading...</div>;
```

Comment:
We should use the Loading Component here?

Reply:
i'll deploy like this now and let's make it nicer later !